fawltydeps/settings.py

```diff
@@ -162,6 +162,7 @@ class Settings(BaseSettings):
     ignore_unused: Set[str] = DEFAULT_IGNORE_UNUSED
     deps_parser_choice: Optional[ParserChoice] = None
     install_deps: bool = False
+    ignores: Set[str] = {".*"}
```

Comment:
Since Maria mentionned it in the dataeng weekly, I'll just add a quick comment that can be safely ignored.

Since FawltyDeps "ignore" can be about different things, namely import of file paths, as a user I may be confused by the name `--ignores`. A more explicit option name (e.g. `--ignore-files`) would be less confusing to me.

Reply:
We have 
- `--ignore-undeclared` to ignore imports that do not have declared dependencies
- `--ignore` for paths we traverse (including files).
So we have an explicit name but for a first case you mentioned. Is this OK?

Reply:
Thanks for raising this question @GuillaumeDesforges. Although we are eager to get this new feature merged and published, this is probably the best time to make sure we get the naming right  (_before_ committing to it in a public release).

FTR, we also have:
- `--ignore-unused` to ignore dependencies that are not (meant to be) imported, such as tools, etc.
- `--ignore-from`: This does not exist yet, but it's what I'm currently working on to support getting `--ignore` pattens indirectly from a file of patterns. To be used like this: `fawltydeps --ignore-from=.gitignore`.

Although all of these options have different names, I cannot prove that this will **NOT** be confusing to some users. There is nothing inherent in the name of  `--ignore` (or `--ignore-from`) that explains that these are about ignoring _paths_. Instead, we hope/assume/believe that users will most easily associate the concept of ignoring _something_ to the paths that are given as input to FawltyDeps. (And if not, we hope that `--help` or the docs will quickly set them straight.)

We could go with something longer, like `--ignore-paths` (and `--ignore-paths-from`?), which would be clearer, but more tedious to type out.

We could also try to re-evaluate our naming and come up with a better scheme. For one: `--ignore-undeclared` and `--ignore-unused` are about ignoring facets of the analysis done by FawltyDeps, but ignoring paths is really about excluding paths from the analysis entirely. As such the path-related options are logically applied/handled at a very different time than the other two. We could consider something like `--exclude`/`--exclude-from` to denote that these are _different_ from the existing `--ignore-undeclared`/`--ignore-unused`? That said, I still find the overlapping of terms across different tools a little clunky here: i.e. that the "gitignore option" is named `--exclude-from=.gitignore` when we have other options called `--ignore-something`.

Maybe we should take a closer look at how other linter tools handle this situation, although I suspect we won't find much consensus...

Reply:
I really like this `--exclude`/`--exclude-from` idea!
It clearly separates what we ignore in traversing directories and what we ignore when showing results. 
In other words `--exclude...` is in the preprocessing step and `--ignore...` in post-processing.

Currently we have one CLI option that uses file as a helper for a setup and it is `--custom-mapping-file`. `--exclude-file` would be extremely confusing, since its purpose it to load a file that defines ignored paths in the gitignore-style. Maybe we should consider having `--custom-mapping-from`. But this would mean backward incompatibility.

Reply:
A quick survey of what other linters do:
- ruff:
  - [`exclude`](https://docs.astral.sh/ruff/settings/#exclude) works on "file patterns" (i.e. adjust analysis input)
  - [`ignore`](https://docs.astral.sh/ruff/settings/#ignore) works on "rule codes" (i.e. adjust result of analysis)
- flake8:
  - [`--exclude`](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-exclude) works on files.
  - [`--ignore`](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-ignore) works on "codes".
- pylint:
  - [`--ignore` (and friends)](https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html#ignore) are used to ignore paths.
  - [`--disable`](https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html#disable) seems to be the main way to disable checkers/rules/codes.
  - Various checkers add other `--ignore-*` options, but these are for ignoring checker-specific details (not related to paths at all).
  - A couple of checkers add `--exclude-*` options, for similar purposes.
  - In general, I don't think pylint is what we should try to emulate. It's interface is a bewildering mess, IMHO.
- black:
  - [`--exclude`](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#exclude): "A regular expression that matches files and directories that should be excluded on recursive searches."
  - Black is famously lacking in options to change its logic/behavior, but there are a few options named [`--skip-*`](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#x-skip-source-first-line) to disable a few parts of its internal logic.
- Mypy:
  - [`exclude`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-exclude) is used to exclude paths
  - For ignoring/disabling internal rules, there is a plethora of options with names like `disallow_*` or `warn_*`.
  - There is not much use of `ignore`, but the few uses that are visible are certainly more related to rules ([`ignore_errors`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-ignore_errors), [`ignore_missing_imports`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-ignore_missing_imports), [`warn_unused_ignores`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-warn_unused_ignores), or [`# type: ignore`](https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#silencing-type-errors)) than to paths/files.

I did not expect to find this much consensus, and it seems a change of opinion is in order:

I now think we should rename `--ignore` (and `--ignore-from`) to `--exclude` (and `--exclude-from`[^1]).

What do you think, @GuillaumeDesforges and @mknorps?

[^1]: Although it's worth having another look at how other tools deal with ignore patterns read from a file like `.gitignore`.

Reply:
(I didn't see your last comment @mknorps before I posted the above survey, but it seems we're in agreement. 🎉)

Re. renaming `--custom-mapping-file` to `--custom-mapping-from`: Yes, it's backwards incompatible, so we would have to keep `...-file` as a deprecated alias to `...-from` for a while, before we remove it completely. Other than that, I don't think it's a problem to rename this.

Reply:
> I now think we should rename --ignore (and --ignore-from) to --exclude (and --exclude-from[1](https://github.com/tweag/FawltyDeps/pull/388#user-content-fn-1-67c636c8d29ad3ceedc5aad716e06f53)).

aye!

Reply:
Makes sense to me :+1: this is probably what triggered my initial comment, somehow my brain had internalized how other tools behave :D

Reply:
Thanks @GuillaumeDesforges for bothering us enough to fix it before we released it! 😅

I've rebased/replaced everything from `ignore` to `exclude` now. I took extra care in the docs to make sure things are as clear and unambiguous as I could make them.